postgresql-persistence/src/main/java/com/netflix/conductor/aurora/AuroraMetadataDAO.java

```diff
@@ -0,0 +1,391 @@
+			.scheduleWithFixedDelay(this::refreshTaskDefs, cacheRefreshTime, cacheRefreshTime, TimeUnit.SECONDS);
+	}
+
+	@Override
+	public String createTaskDef(TaskDef taskDef) {
+		validate(taskDef);
+		if (null == taskDef.getCreateTime() || taskDef.getCreateTime() < 1) {
+			taskDef.setCreateTime(System.currentTimeMillis());
+		}
+
+		return insertOrUpdateTaskDef(taskDef);
+	}
+
+	@Override
+	public String updateTaskDef(TaskDef taskDef) {
+		validate(taskDef);
+		taskDef.setUpdateTime(System.currentTimeMillis());
+		return insertOrUpdateTaskDef(taskDef);
+	}
+
+	@Override
+	public TaskDef getTaskDef(String name) {
+		Preconditions.checkNotNull(name, "TaskDef name cannot be null");
+		TaskDef taskDef = taskDefCache.get(name);
+		if (taskDef == null) {
+			if (logger.isTraceEnabled()) {
+				logger.trace("Cache miss: {}", name);
+			}
+			taskDef = getTaskDefFromDB(name);
+		}
+
+		return taskDef;
+	}
+
+	@Override
+	public List<TaskDef> getAllTaskDefs() {
+		return getWithTransaction(this::findAllTaskDefs);
+	}
+
+	@Override
+	public void removeTaskDef(String name) {
+		final String SQL = "DELETE FROM meta_task_def WHERE name = ?";
+
+		executeWithTransaction(SQL, q -> {
+			if (!q.addParameter(name).executeDelete()) {
+				throw new ApplicationException(ApplicationException.Code.NOT_FOUND, "No such task definition");
+			}
+
+			taskDefCache.remove(name);
+		});
+	}
+
+	@Override
+	public void create(WorkflowDef def) {
+		validate(def);
+		if (null == def.getCreateTime() || def.getCreateTime() == 0) {
+			def.setCreateTime(System.currentTimeMillis());
+		}
+
+		withTransaction(tx -> {
+			if (workflowExists(tx, def)) {
+				throw new ApplicationException(ApplicationException.Code.CONFLICT,
+					"Workflow with " + def.key() + " already exists!");
+			}
+
+			insertOrUpdateWorkflowDef(tx, def);
+		});
+	}
+
+	@Override
+	public void update(WorkflowDef def) {
+		validate(def);
+		def.setUpdateTime(System.currentTimeMillis());
+		withTransaction(tx -> insertOrUpdateWorkflowDef(tx, def));
+	}
+
+	@Override
+	public void removeWorkflow(WorkflowDef def) {
+		validate(def);
+		String name = def.getName();
+		int version = def.getVersion();
+		final String SQL = "DELETE from meta_workflow_def WHERE name = ? AND version = ?";
+
+		executeWithTransaction(SQL, q -> {
+			if (!q.addParameter(name).addParameter(version).executeDelete()) {
+				throw new ApplicationException(ApplicationException.Code.NOT_FOUND,
+					String.format("No such workflow definition: %s version: %d", name, version));
+			}
+		});
+	}
+
+	@Override
+	public WorkflowDef getLatest(String name) {
+		final String SQL = "SELECT json_data FROM meta_workflow_def WHERE NAME = ? AND " +
+			"version = latest_version";
+
+		return queryWithTransaction(SQL,
+			q -> q.addParameter(name).executeAndFetchFirst(WorkflowDef.class));
+	}
+
+	@Override
+	public WorkflowDef get(String name, int version) {
+		final String SQL = "SELECT json_data FROM meta_workflow_def WHERE NAME = ? AND version = ?";
```

Comment:
Why aren't the workflow definitions cached like the task definitions?

Reply:
Good point. We should consider to add that but before, need to evaluate workflow definition size in mb which will be sitting in the memory.

Reply:
Makes sense.